cpp/tests/sampling/mg_negative_sampling.cu

```diff
@@ -0,0 +1,411 @@
+  Tests_MGNegative_Sampling<cugraph::test::Rmat_Usecase, int64_t, int64_t, float>;
+
+template <typename CurrentTest>
+void run_all_tests(CurrentTest* current_test)
+{
+  raft::random::RngState rng_state{
+    static_cast<uint64_t>(current_test->handle_->get_comms().get_rank())};
+
+  current_test->run_current_test(
+    rng_state, Negative_Sampling_Usecase{2, false, false, false, false, false, true});
+  current_test->run_current_test(
+    rng_state, Negative_Sampling_Usecase{2, true, false, false, false, false, true});
+  current_test->run_current_test(
+    rng_state, Negative_Sampling_Usecase{2, false, true, false, false, false, true});
+  current_test->run_current_test(
+    rng_state, Negative_Sampling_Usecase{2, true, true, false, false, false, true});
+  current_test->run_current_test(
+    rng_state, Negative_Sampling_Usecase{2, false, false, true, false, false, true});
+  current_test->run_current_test(
+    rng_state, Negative_Sampling_Usecase{2, true, false, true, false, false, true});
+  current_test->run_current_test(
+    rng_state, Negative_Sampling_Usecase{2, false, true, true, false, false, true});
+  current_test->run_current_test(
+    rng_state, Negative_Sampling_Usecase{2, true, true, true, false, false, true});
+  current_test->run_current_test(
+    rng_state, Negative_Sampling_Usecase{2, false, false, false, true, false, true});
+  current_test->run_current_test(
+    rng_state, Negative_Sampling_Usecase{2, true, false, false, true, false, true});
+  current_test->run_current_test(
+    rng_state, Negative_Sampling_Usecase{2, false, true, false, true, false, true});
+  current_test->run_current_test(
+    rng_state, Negative_Sampling_Usecase{2, true, true, false, true, false, true});
+  current_test->run_current_test(
+    rng_state, Negative_Sampling_Usecase{2, false, false, true, true, false, true});
+  current_test->run_current_test(
+    rng_state, Negative_Sampling_Usecase{2, true, false, true, true, false, true});
+  current_test->run_current_test(
+    rng_state, Negative_Sampling_Usecase{2, false, true, true, true, false, true});
+  current_test->run_current_test(rng_state,
+                                 Negative_Sampling_Usecase{2, true, true, true, true, false, true});
+  current_test->run_current_test(
+    rng_state, Negative_Sampling_Usecase{2, false, false, false, false, true, true});
+  current_test->run_current_test(
+    rng_state, Negative_Sampling_Usecase{2, true, false, false, false, true, true});
+  current_test->run_current_test(
+    rng_state, Negative_Sampling_Usecase{2, false, true, false, false, true, true});
+  current_test->run_current_test(
+    rng_state, Negative_Sampling_Usecase{2, true, true, false, false, true, true});
+  current_test->run_current_test(
+    rng_state, Negative_Sampling_Usecase{2, false, false, true, false, true, true});
+  current_test->run_current_test(
+    rng_state, Negative_Sampling_Usecase{2, true, false, true, false, true, true});
```

Comment:
@seunghwak - When I run this test with np = 2 I'm getting the error in shuffling (np = 1 works fine).  Haven't tried disabling just this test to see if any tests after this fail.

Reply:
Fixed